src/_pytest/compat.py

```diff
@@ -210,29 +209,15 @@ def ascii_escaped(val: bytes | str) -> str:
     return ret.translate(_non_printable_ascii_translate_table)
 
 
-@dataclasses.dataclass
-class _PytestWrapper:
-    """Dummy wrapper around a function object for internal use only.
-
-    Used to correctly unwrap the underlying function object when we are
-    creating fixtures, because we wrap the function object ourselves with a
-    decorator to issue warnings when the fixture function is called directly.
-    """
-
-    obj: Any
-
-
 def get_real_func(obj):
     """Get the real function object of the (possibly) wrapped object by
-    functools.wraps or functools.partial."""
+    functools.wraps or functools.partial or pytest.fixture"""
+    from _pytest.fixtures import FixtureFunctionDefinition
```

Comment:
This is cyclic if I remember correctly. I will try to move it to a another file and try to resolve it

Reply:
[I tried](https://github.com/pytest-dev/pytest/pull/12473/commits/f8167dbbb9fad2099d7fa242438eed7ba9e49774) moving it to fixture.py file so we can get rid of the import(otherwise it's cylcic) but that would require a lot more refactoring. I prefer to leave the import here or introduce a new file and put this function or FixtureFunctionDefinition there. What do you think?